### DISCUSSION TITLE
SDK to interact with PocketHost

### DISCUSSION THREAD

**Discussion by charbs-io** [pockethost/pockethost#275] [Q&A]
Hey @benallfree ,

Are you at all thinking or planning on providing an SDK? I see some roadmap items relating to CLI but no mention of an SDK.

The usecase I have in mind (and this is a pie in the sky thought at the moment) is a non-technical CMS allowing users to create complex apps and websites in functionality without the need to commit code (think all the great things we get with PocketBase for free like Auth etc). The service would internally interface with some sort of a PocketHost SDK to create instances etc. 

Commercially this would play well with the pricing models i've seen in some discussions, meaning the owner of the account spinning up instances would be responsible for paying for their use of PocketHose and potentially passing some of that cost to their own clients/users. PocketHost would just be responsible for hosting these instances so long as it provides a way to delete instances (I know that deleting instances is an open issue at the moment). 

Thanks!

**Comment by benallfree** [comment]
@charbs-io I'm curious about the use case here, what would be the purpose of programmatically provisioning and managing instances? The CLI project would necessitate an API, just curious what other ideas you had in mind.
